Extensions/DataSources/src/DataSourcesHub/components/DataSourceDetails.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import { Header, TitleSize } from "azure-devops-ui/Header";
+import { TextField, TextFieldWidth } from "azure-devops-ui/TextField";
+import { DataSourcesActionCreators } from "../action-creators/DataSourcesActionCreators";
+import { DATA_SOURCE } from '../Resources/DataSourcesResources';
+import { ParamInfo, DataSourceInfo } from "../states/DataSourcesExtensionState";
+
+type DataSourceDetailsProps= {
+    currentInputParam: ParamInfo | null
+    displayInfo: string | null  
+    datasourcesInfo : DataSourceInfo | null 
+}
+
+export class DataSourceDetails extends React.Component<DataSourceDetailsProps>{
+    public onChange(e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, item:string) {
+        if(this.props.datasourcesInfo!=null){
+            let DataSourceActionCreator = new DataSourcesActionCreators(null);
+            DataSourceActionCreator.updateDataSource(item,this.props.currentInputParam);
+        }
+    }
+    
+    public render() {
+        if(this.props.displayInfo!=null){
+            return(
+                <div>
+                    <div className="datasource-detail-vertical-spacing-16">
```

Comment:
do not name your class based on the css you are applying
it should just be datasource-detail

please fix all css class names